app/Chatkit Quickstart/ChatroomViewController.swift

```diff
@@ -58,96 +54,101 @@ class ChatroomViewController: UIViewController {
         messagesTableView.dataSource = viewModel
         messagesTableView.allowsSelection = true
         
+        // Load Chatkit instance details from the plist file
+        let chatkitInfo = plistValues(bundle: Bundle.main)
+        
         // Instantiate Chatkit with instance ID, token provider endpoint, and ID of the user you will connect as.
         // Initialization: https://pusher.com/docs/chatkit/reference/swift#initialization
         // Authenticating users and and providing tokens: https://pusher.com/docs/chatkit/reference/swift#pctokenprovider
-        chatManager = ChatManager(
+        let chatManager = ChatManager(
             instanceLocator: chatkitInfo.instanceLocator,
             tokenProvider: PCTokenProvider(url: chatkitInfo.tokenProviderEndpoint),
             userID: chatkitInfo.userId
         )
+        self.chatManager = chatManager
         
         // Connect to Chatkit by passing in the ChatManagerDelegate defined at the top of this class.
         // https://pusher.com/docs/chatkit/reference/swift#connecting
-        chatManager!.connect(delegate: MyChatManagerDelegate()) { (currentUser, error) in
-            guard(error == nil) else {
+        chatManager.connect(delegate: ChatManagerDelegate()) { (currentUser, error) in
+            guard error == nil else {
                 print("Error connecting: \(error!.localizedDescription)")
                 return
             }
-            
-            DispatchQueue.main.async {
```

Comment:
I think this is still required, the callback won't be made on the main thread, and even if we don't touch any UI components directly, none of the controller methods are really threadsafe...

Reply:
This sounds sensible, especially knowing how highly concurrent and barely synched v1 is.

Reply:
I have now added the `DispatchQueue.main.async` back :-)